### PR TITLE
add a job that passes only if all others pass

### PR DESCRIPTION
**Pull request by bengl**

### CODE REVIEW

File: .github/workflows/project.yml [hunk 1]
```diff
@@ -112,3 +112,13 @@ jobs:
...
       - run: yarn type:test
       - run: yarn type:doc
+
+  all-green:
+    runs-on: ubuntu-latest
+    permissions:
+      checks: read
+      contents: read
+    steps:
+      - uses: wechuli/allcheckspassed@v1
```
Comment (rochdev) [inline .github/workflows/project.yml:122]: Why is this needed? GHA has this built-in, and it's instant instead of having to poll the status of the project (which can cause other problems like hitting rate limits, which is probably why this action is very slow)

Comment (bengl) [reply .github/workflows/project.yml:122]: > Why is this needed? GHA has this built-in, and it's instant instead of having to poll the status of the project

AFAICT, there's no way of requiring that _all_ checks have passed, _regardless_ of whether new ones have been added or whatever. AFAIK, there are only two ways to ensure everything is green all the time.

### Option 1

Manually edit this list every time a new check is added. For projects with small amounts of checks, this is fine, but we currently have many dozens of them and are adding new ones all the time.

Screenshot 2024-05-22 at 12 06 39 PM

### Option 2

Use something like this which polls the API. 

> (which can cause other problems like hitting rate limits, which is probably why this action is very slow)

This action is [configurable](https://github.com/wechuli/allcheckspassed?tab=readme-ov-file#examples). Currently I have it set to poll once a minute for 20 minutes, or until all jobs are either passed, failed, or cancelled/stopped. We can change this to whatever we want. I chose 20 min because our longest checks (e.g. system tests and the next.js test) take something like 15 min. Because it waits for every other one, but is triggered at the same time, this will always be the longest one. [We _can_ configure the job to only be triggered after a few min](https://github.com/wechuli/allcheckspassed?tab=readme-ov-file#setup-with-environments), so that we reduce its total running time.

Comment (rochdev) [reply .github/workflows/project.yml:122]: I missed that this also verifies any external non-GHA status checks. In that context it makes sense that polling the project is needed.

**Review by rochdev — APPROVED** [review]
